### PR TITLE
Update to NixOS 23.05

### PR DESCRIPTION
**Pull request by erikarvstedt**
Things to do:
- [x] Check if the current `twisted` version works correctly with joinmarket.
       twisted 22.4.0 (the version explicitly supported by joinmarket) is very cumbersome to build with NixOS 23.05.
- [x] Wait for https://github.com/systemd/systemd/issues/28030 to be fixed.
  This currently breaks our tests. See commit `[tmp] add systemd patch`.

### CODE REVIEW

**Comment by erikarvstedt** [comment]
We can update to 23.05 when the last item ([`nixos-23.05`](https://hydra.nixos.org/job/nixos/release-23.05/tested#tabs-constituents)) in [this progress tracker](https://nixpk.gs/pr-tracker.html?pr=238971)  is green.

**Comment by erikarvstedt** [comment]
The CI failure is caused by https://github.com/cachix/cachix/issues/547.

**Comment by erikarvstedt** [comment]
Ready to merge. This branch is running live [on nixbitcoin.org](https://github.com/erikarvstedt/nixbitcoin.org/commits/nixos-23.05).

When releasing, we need to mention the pkg updates from commit `update nixpkgs`:
clightning: 23.05.1 -> 23.05.2
lightning-pool: 0.5.3-alpha -> 0.6.4-beta

**Comment by erikarvstedt** [comment]
I've extracted some patches to https://github.com/fort-nix/nix-bitcoin/pull/625.

File: .cirrus.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ task:
 
   container:
     # Defined in https://github.com/nix-community/docker-nixpkgs
+    # TODO-EXTERNAL:
+    # Update to nixos-23.05 when this issue has been resolved:
+    # https://github.com/nix-community/docker-nixpkgs/issues/60
     image: nixpkgs/nix-flakes:nixos-22.11
```
Comment (jonasnick) [inline .cirrus.yml:15]: seems like this has just been resolved

File: pkgs/lib.nix [hunk 2]
```diff
@@ -41,8 +41,13 @@ let self = {
...
   };
 
-  # nodejs applications require memory write execute for JIT compilation
-  nodejs = { MemoryDenyWriteExecute = false; };
+  nodejs = {
+    # Required for JIT compilation
+    MemoryDenyWriteExecute = false;
+    # Required by nodejs >= 18
+    # TODO-EXTERNAL: Remove this when `pkey_alloc` has been added to systemd's `@system-service`
+    SystemCallFilter = self.defaultHardening.SystemCallFilter ++ [ "pkey_alloc" ];
```
Comment (jonasnick) [inline pkgs/lib.nix:49]: I also need to add "pkey_mprotect" here. Otherwise, `clightning-rest` crashes with a coredump in the default test scenario.

**Review by jonasnick — APPROVED** [review]
ACK e658209d568573a5811c1696de9cffafc099d045

**Pull request merged**